src/cpp/src/utils.cpp

```diff
@@ -265,6 +267,45 @@ ov::Core singleton_core() {
     return core;
 }
 
+size_t get_first_history_difference(const ov::Tensor& encoded_history, const std::vector<int64_t> tokenized_history) {
+    size_t i = 0;
+    auto encoded_history_data = encoded_history.data<int64_t>();
+    while(i < encoded_history.get_size()) {
+        if (i >= tokenized_history.size() || encoded_history_data[i] != tokenized_history[i])
+            break;
+        i++;
+    }
+
+    return i;
+}
+
+bool is_tokenized_history_same(const ov::Tensor& encoded_history, const std::vector<int64_t> tokenized_history) {
+    // encoded_history should be eq or less then tokenized_history, tokenized_history also include eos token
+    if (encoded_history.get_size() > tokenized_history.size())
+        return false;
+    size_t idx = get_first_history_difference(encoded_history, tokenized_history);
+
+    // encoded_history after decode of tokenizer could lose one last token (eos/stop token)
+    return idx == tokenized_history.size() || idx == tokenized_history.size() - 1;
```

Comment:
should we check that last token of encoded history is EOS or another stop token for the second condition ?

Reply:
just not sure about stop tokens, it is setup parameter for model/sampling, but tokenizer will operate in the decoder/encode with its eos token